### PR TITLE
Adding Lattice

### PR DESCRIPTION
**Pull request by cgentry7**
This adds the Lattice class.

### CODE REVIEW

File: test/unit/test_lattice.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+def test_lattice_write_to_string(lattice, module):
+    output = lattice.write_to_string(prefix="  ",
+                                     module_mpact_ids={module: 2},
+                                     lattice_mpact_ids={lattice: 4})
+    expected_output = "  lattice 4 2 2\n" + \
+                      "    2 2\n" + \
+                      "    2 2\n"
+    assert output == expected_output
+
+def test_lattice_get_axial_slice(lattice):
```
Comment (npl436) [inline test/unit/test_lattice.py:52]: My usual comment.  You could also have a test for invalid input for `get_axial_slice` when it returns `None`.

**Review by npl436 — APPROVED** [review]
Thank you.  I will merge.  Please take or leave my comments as you see fit.

File: mpactpy/lattice.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        a 2-D array of modules
+    """
+
+    class Pitch(TypedDict):
+        """ A Typed Dictionary class for Lattice Pitches
+        """
+        X: float
+        Y: float
+        Z: float
+
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/lattice.py:37]: We might do this eventually, but I'd have to think this through more.  Yes, there is some overlap, but there are differences in parameters for `write_to_string` and I think it's good to maintain the different contexts of things like `Pitch` and `ModDim`.  For example, `ModDim` for `Lattice` means something different than `ModDim` for `Assembly`.

In short, we could probably kill some code redundancy here, but I think there's just enough difference between the classes that it's probably better to leave as-is for now.

File: mpactpy/lattice.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    class ModDim(TypedDict):
+        """ A Typed Dictionary class for Lattice Module Dimensions
+        """
+        X: float
+        Y: float
+        Z: float
+
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/lattice.py:44]: From a typehinting perspective, I like having these two a separate `TypedDict` rather than the same.  There's a little redundancy, but I want to drive home the point that `ModDim` shouldn't be treated as equivalent to `Pitch`.  `Pitch` corresponds the full lattice pitch, whereas `ModDim` corresponds to the module pitch.

File: mpactpy/module.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            for pin in row:
+                stack = None
+                for z in range(self.nz):
+                    z0 = z * pin.pitch["Z"]
+                    pin_slice = pin.get_axial_slice(start_pos-z0, stop_pos-z0)
+                    if pin_slice:
+                        stack = stack.axial_merge(pin_slice) if stack else pin_slice
+                pin_map[-1].append(stack)
+
+        return Module(1, pin_map)
```
[the remark this answers is not in the document]
Comment (cgentry7) [reply mpactpy/module.py:170]: Added in #22.  Can't think of an immediate use for it, but I could definitely see it having potential.

File: test/unit/test_lattice.py (hunk 1 above)
Comment (cgentry7) [reply test/unit/test_lattice.py:52]: Done in #22
